lectures/MobilePlatforms/ReadMe.md

```diff
@@ -0,0 +1,176 @@
+
+To understand fragments, please see [the Android documentation](https://developer.android.com/guide/fragments).
+
+You don't have to use fragments within activities, but such modularization makes the app more flexible and also makes it easier to maintain over time. A fragment defines and manages its own layout, has its own lifecycle, and can handle its own input events. But it cannot live on its own, it needs to be hosted by an activity or another fragment.
+
+Android beginners tend to put all their logic inside Activities and Fragments, ending up with "views" that do a lot more than just render the UI. Remember what we said about MVVM and how that maps to how you build an Android app. In the exercise set, we will ask you to go through an MVVM exercise. Android provides a nice ViewModel class to store and manage UI-related data.
+
+## User experience (UX)
+
+User experience design (or "UX design") is about putting together an intuitive, responsive, navigable, usable interface to the app. You want to look at the app through the users' perspective to derive what can give them an easy, logical and positive experience.
+
+So you must ask who is your audience: Old or young people? Intellectuals or blue-collar workers? Think back to the lecture on personas and user stories.
+
+The UI should enable the user to get around the app easily, and to find quickly what they're looking for. To achieve this, several elements and widgets have emerged from the years of experience of building mobile apps. The [hamburger icons](https://en.wikipedia.org/wiki/Hamburger_button) are used for drop-down menus with further details--they avoid clutter. Home buttons give users a shortcut to home base. Chat bubbles offer quick help through context-sensitive messages.
+
+Personalization of the UI allows adapting to what the user is interested in, keeping the unrelated content away. For example, in the [EPFL Campus app](https://pocketcampus.org/epfl-en) you can select which features you see on the home screen, and so the home screen of two different users is likely to look different.
+
+In order to maximize readability, emphasize simplicity and clarity, choose adequate font size and image size. Avoid having too many elements on the screen, because that leads to confusion. Offer one necessary action per screen. Use [micro UX animations](https://uxplanet.org/ui-design-animations-microinteractions-50753e6f605c), which are little animations that appear when a specific action is performed or a particular item is hovered or touched; this can increase engagement and interactivity. E.g., hovering over a thumbnail shows details about it. E.g., hover zoom for maximizing the view of a specific part of something.  Keep however the animations minimal, avoid flashiness, make them subtle and clear as to their intent.
```

Comment:
I would add something about remaining consistent with Android itself. Users expect specific looks for specific UI elements, for instance.

Reply:
Also, "hover" is a bit obscure in the context of Android, very few devices have hovering capabilities (only tablets with pens, off the top of my head?)

Reply:
About Android, yes, good point. Are there some specific elements to mention, that pretty much all Android apps exhibit?

Reply:
Ah yes the two-level touch thing is I think iOS-only.

Re: Android, a hamburger menu is probably a good example, along with the general fact that Android offers existing controls and users shouldn't reinvent their for, e.g., displaying a list of items, nor should they override the default margins, colors, etc. without a good reason.